### ISSUE TITLE
Add Webhook JSON customizability or Slack WebHook format support

### ISSUE THREAD

**Issue by mkorkalo** [sullivanmatt/bugalert#24]
Currently there's no easy to to integrate with slack using the standard methods.
https://api.slack.com/messaging/webhooks

I see two logical options:

1. Allow to customize the webhook json payload e.g. editable json template with fields as ${parameters}, or similar. This has the upside of supporting many APIs, including Slack's and customizing message can be done. Downside is you're possibly creating a way for users to use the site to post custom requests anywhere. Maybe that's not a big deal?

2. If you do not wish to allow custom payload, maybe just add an option for slack payload format and create a default slack template that should be good enough for anyone.

**Comment by sullivanmatt** [comment]
This will definitely take some dev work, but I think it's a great idea to parameterize our values and let users customize the JSON body, possibly with some templates for major services to help ease the burden for the average user.

I also need to build an ability to test webhooks anyway, so this area will get some much needed love soon.

**Comment by mkorkalo** [comment]
Awesome! I thought I'd better speak out now when the feature is still in development.

**Comment by mattlorimor** [comment]
> Currently there's no easy to to integrate with slack using the standard methods.

Yeah, until something like this exists, the easiest stopgap solution is probably to get the [@BugAlertDotOrg](https://twitter.com/BugAlertDotOrg) Tweets to a Slack channel.

**Comment by pointydev** [comment]
Something to consider if offering premade payloads would be Discord's [webhook format](https://discord.com/developers/docs/resources/webhook), as well as their handling of [Slack-compatible](https://discord.com/developers/docs/resources/webhook#execute-slackcompatible-webhook) webhooks.
